build_script.py

```diff
@@ -181,15 +195,34 @@ def build(args):
                 core_foundation_build_dir=core_foundation_build_dir,
                 libdispatch_args=libdispatch_args,
                 source_paths=" ".join(sourcePaths)))
+
+        xctest_lib = 'libXCTest.so'
+        xctest_soname = '{}.{}.{}.{}'.format(xctest_lib, *VERSION)
+
         run("{swiftc} -emit-library {build_dir}/XCTest.o "
             "-L {foundation_build_dir} -lswiftGlibc -lswiftCore -lFoundation -lm "
             # We embed an rpath of `$ORIGIN` to ensure other referenced
             # libraries (like `Foundation`) can be found solely via XCTest.
             "-Xlinker -rpath=\\$ORIGIN "
-            "-o {build_dir}/libXCTest.so".format(
+            "-Xlinker -soname={soname} "
+            "-o {build_dir}/{soname} "
+            "-module-name XCTest".format(
                 swiftc=swiftc,
                 build_dir=build_dir,
-                foundation_build_dir=foundation_build_dir))
+                foundation_build_dir=foundation_build_dir,
+                soname=xctest_soname))
+
+        # create symlinks for versions
+        with SetDir(build_dir):
+            # symlink lib.so
+            if not os.path.exists(xctest_lib):
+                note('ln -s {} {}'.format(xctest_soname, xctest_lib))
```

Comment:
`os.symlink` creates a symlink. Could we use that instead? `ln -s` would not work on Windows. I understand that Windows is a long way off, but it doesn't hurt to be prepared. 😄